.buildkite/pipelines/build_linux.json.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+#
+# This script generates a build pipeline for linux.
+#
+# OUTPUT:
+#   - linux build JSON file.
+#
+
+import argparse
+import json
+import os
+
+from itertools import product
+
+archs = [
+    "x86_64",
+    "aarch64",
+]
+build_types = [
+    "RelWithDebInfo",
+]
+actions = [
+    "build",
+    "debug"
+]
+agents = {
+   "x86_64": {
+      "cpu": "6",
+      "ephemeralStorage": "20G",
+      "memory": "64G",
+      "image": "docker.elastic.co/ml-dev/ml-linux-build:14"
+   },
+   "aarch64": {
+      "provider": "aws",
+      "instanceType": "m6g.2xlarge",
+      "imagePrefix": "ci-amazonlinux-2-aarch64",
+      "diskSizeGb": "100",
+      "diskName": "/dev/xvda"
+   }
+}
+
+def main(args):
+    pipeline_steps = []
+    cur_build_types = build_types
+    if args.build_type is not None:
+        cur_build_types = [args.build_type]
+
+    pipeline_steps.append({
+        "label": "Clone Eigen",
+        "key": "clone_eigen",
+        "commands": [
+            './3rd_party/pull-eigen.sh',
+            'tar cvzf eigen.tgz 3rd_party/eigen',
+            'buildkite-agent artifact upload eigen.tgz'
+            ]
+        })
+    for arch, build_type in product(archs, cur_build_types):
+        pipeline_steps.append({
+            "label": f"Build & test :cpp: for linux-{arch}-{build_type} :linux:",
+            "timeout_in_minutes": "150",
+            "agents": agents[arch],
+            "commands": [
+              f'if [[ "{args.action}" == "debug" ]]; then export ML_DEBUG=1; fi',
+              'buildkite-agent artifact download eigen.tgz .',
+              'tar xvzf eigen.tgz',
+              ".buildkite/scripts/steps/build_and_test.sh"
+            ],
+            "depends_on": [
+              "check_style",
+              "clone_eigen"
+              ],
+            "key": f"build_test_linux-{arch}-{build_type}",
+            "env": {
+              "CMAKE_FLAGS": f"-DCMAKE_TOOLCHAIN_FILE=cmake/linux-{arch}.cmake",
```

Comment:
7.17 doesn't use CMake - but is something else needed instead?

Reply:
Thanks Dave, I think these `CMake` references can simply be removed as there is no equivalent configuration in the GNU make system.